app/jobs/check_fixity_recursive_job.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+class CheckFixityRecursiveJob < ApplicationJob
+  delegate :query_service, to: :metadata_adapter
+
+  def perform
+    file_set = find_next_file_to_check
+    original_file_metadata = file_set.run_fixity
+    file_set.file_metadata = file_set.file_metadata.select { |x| !x.original_file? } + Array.wrap(original_file_metadata)
+    metadata_adapter.persister.save(resource: file_set)
+    CheckFixityRecursiveJob.set(queue: queue_name).perform_later
+  end
+
+  private
+
+    def find_next_file_to_check
+      query_service.find_all_of_model(model: FileSet).map(&:decorate).sort_by(&:fixity_sort_date).first
```

Comment:
Can we avoid loading and decorating all the FileSets?  We have ~3M FileSets now, and that will keep growing, so I'm worried about how well this will perform.

Reply:
I don't really know what sorts of methods will load them and what won't. some guidance here would be appreciated.

Reply:
We'll probably need a custom query for this that runs some SQL. We might need a custom index to make it sortable though..

Reply:
hm. okay that `fixity_sort_date` method holds some logic that draws on multiple attributes; not sure whether that will translate to an SQL query. maybe? geez, it's been a long time since I did any SQL.

Reply:
I believe this is the SQL we need to get the oldest fixity_last_run_date (never checked at the top), sub-sorted by created date:

```
select * from orm_resources
order by metadata->>'fixity_last_run_date' nulls first, metadata->>'created_at'
```

Reply:
nice, thanks @escowles! it's slightly different logic but I don't have any compelling argument for the way I did it in light of performance concerns. I'll try this out.

Reply:
Seems like I could add `limit 1` to the end of that, also, since i just need one at a time

Reply:
We'll need some indexes on fixity_last_run_date and created_at to make that perform, I think. Running it on prod now to see..